### PR TITLE
Artifact selling and classification

### PR DESCRIPTION
**Pull request by zjdtmkhzt**
## About the PR 
Adds some form type stuff that you can use to describe artifacts, specifically their origin, type, and activation method.
There's also some freeform fields for describing faults and other stuff.
They work like stickers, if you slap one on an artifact and all the info is correct, it'll make the artifact have the real name. 
(They will no longer have the real name just from being activated.)

You can also sell artifacts via cargo now, how much money you get depends on the rarity of the artifact and whether or not it has a sticker with the correct info about it stuck to it.
Prices approximately: Tier 1, 300; Tier 2, 900; Tier 3, 4.000; Tier 4, 12.000
(These prices are when fully analyzed, when unanalyzed it is only a third.)
If you're lucky, you might also get a new artifact sent in at cargo a few minutes after having one sent out.
(This also scales with the rarity of the artifacts that were sent out and the accuracy of the information in the sheet.)

Beware the terrible codersprites and also terrible coder-ui.
image

## Why's this needed? 
- I find that documenting information about artifacts gives the artlab even more of a "sciencey" feel.
- Selling artifacts is a feature that I often see requested.
- This will give at least some use to stuff like lamps/gravity wells/noisemakers.
- The artlab getting more artifacts from cargo and cargo getting money from the artlab should foster interdepartmental cooperation.
- It gives more options for the artlab to get more artifacts without having to bugger off to space when mining isn't very active and without having to metagame all the guaranteed artifact spawn locations.
## Changelog

```changelog
(u)zjdtmkhzt
(*)Artifacts are now sellable, more details below in the minor entries.
(+)Money depends on artifact rarity.
(+)Artlab has these forms you can fill out. If you fill it out correctly and slap it on an artifact, you'll get a lot more money for it.
(+)Sometimes cargo might get new artifacts sent in if they previously sent nice ones out.
```

### CODE REVIEW

**Comment by zjdtmkhzt** [comment]
I still need to polish the UI a little and add the form dispensers to the other maps, but since the code is basically done, I just thought I'd put this up to get feedback in the meantime.

**Comment by Boxlta** [comment]
Vry good, looks cool, Selling arts is one thing that felt like it could be a cool system.

**Comment by ZeWaka** [comment]
what is the 'popup thing'

**Comment by Azrun** [comment]
Seems like a number of the artifact types can be condensed (bombs, wand/non-wand, power cells) from a player perspective.  Bomb types seem to be differentiated based on xray feedback which seems like it would require previous experience or a code dive?  I’d expect groups similar to wiki.  Being able to reward for the differentiation is neat though, seems like a great use of additional info but would then require _PARSING_.

Is the name “Precursor” appropriate?  Outside of the wiki didn't really find anything that called them out as such, but my lore is weak.

Really like that most of this is generated programmatically. 👨‍💻 

Might be neat to roll in lastAnalysis correctness into how crate is received.  Maybe improve/adjust delay or rarity of artifacts.  “You have been identified as a critical artifact researcher! You have been rewarded with more work!”

File: tgui/packages/tgui/interfaces/ArtifactPaper.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    <Window
+      resizable
+      title="Nanotrasen Alien Artifact Analysis Form"
+      theme="paper"
+      width={800}
+      height={825}
+    >
+      <Section>
+        <h3>Artifact Name</h3>
+        <h4>{artifactName===""?"unknown":artifactName}</h4>
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/ArtifactPaper.js:29]: spacing please

File: tgui/packages/tgui/interfaces/ArtifactPaper.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                checked={artifactTriggers === x}
+              />
+              <a>{x}</a>
+            </Flex.Item>
+          ))}
+        </Flex>
+        <h3>Artifact Faults</h3>
+        <TextArea
+          value={artifactFaults}
+          backgroundColor="#DDDDDD"
```
Comment (ZeWaka) [inline tgui/packages/tgui/interfaces/ArtifactPaper.js:69]: hardcoded colors boo

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by zjdtmkhzt** [comment]
Hmm, I guess they are never explicitly referred to as Precursors in the game, but they aren't really ever referred to anything ingame.
The wiki calls them Forerunners, but that title is absolutely never used ingame, I think, and it does mention that Precursor is another name for them.

Honestly, I always call them Precursor ingame, seeing as there's no actual name that is ever given and Precursor sounds way cooler. I think people will figure it out, based on the wiki?

File: code/obj/artifacts/artifactanalysis.dm [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+		if(A.type_name == src.artifactType)
+			lastAnalysis++
+
+		// check if trigger is one of the correct ones
+		for(var/datum/artifact_trigger/T as() in A.triggers)
+			if(T.type_name == src.artifactTriggers)
+				lastAnalysis++
+				break
+		// if a trigger would e redundant, let's just say it's cool!
+		if(length(A.triggers) == 0 || A.automatic_activation)
```
[the remark this answers is not in the document]
Comment (zjdtmkhzt) [reply code/obj/artifacts/artifactanalysis.dm:36]: I considered using !, but seeing as it's not a boolean kind of thing but actually an integer, I felt using a proper comparison would be clearer?

[the remark this answers is not in the document]
Comment (zjdtmkhzt) [reply code/obj/artifacts/artifactanalysis.dm:36]: But I guess I don't mind changing it either.

**Comment by zjdtmkhzt** [comment]
That reminds me, I should probably use the nice names for the bomb alerts, now that we have them.

**Review by ZeWaka — APPROVED** [review]

**Comment by zjdtmkhzt** [comment]
Added to all the player pickable maps except Clarion now.

**Pull request merged**